src/wechaty_puppet/file_box/file_box.py

```diff
@@ -109,12 +143,39 @@ def to_json_str(self) -> str:
         data = json.dumps(json_data, cls=FileBoxEncoder, indent=4)
         return data
 
-    def to_file(self, file_path: str) -> None:
+    async def to_file(self, file_path: Optional[str] = None,
+                      overwrite: bool = False):
         """
         save the content to the file
         :return:
         """
-        raise NotImplementedError
+        if self.type() == FileBoxType.Url:
+            if not self.mimeType or not self.name:
+                await self.sync_remote_name()
+
+        file_path = self.name if file_path is None else file_path
+
+        if os.path.exists(file_path) and not overwrite:
+            raise FileExistsError(f'FileBox.toFile(${file_path}): file exist. '
+                                  f'use FileBox.toFile(${file_path}, true) to '
+                                  f'force overwrite.')
+        file_box_type = self.type()
+
+        if file_box_type == FileBoxType.Buffer:
+            with open(file_path, 'wb+', encoding='utf-8') as f:
```

Comment:
this line will raise` internal error <binary mode doesn't take an encoding argument>`

Reply:
Oh yes, if the mode of IoStream is 'wb', we should not set encoding argument.